pytket_qir/qir.py

```diff
@@ -400,11 +445,80 @@ def condition_zero_block():
                 one=lambda: condition_one_block(),
                 zero=lambda: condition_zero_block(),
             )
+        elif isinstance(op, WASMOp):
+            cregs = _retrieve_registers(circ.bits, BitRegister)
+            args = command.args
+
+            inputs: List[str] = []
+            outputs: List[str] = []
+            for reglist, sizes in [
+                (inputs, op.input_widths),
+                (outputs, op.output_widths),
+            ]:
+                for in_width in sizes:
+                    wasm_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = wasm_bits[0].reg_name
+                    if wasm_bits != list(cregs[regname]):
+                        WASMUnsupportedError("WASM ops must act on entire registers.")
+                    reglist.append(regname)
+
+            reg = circ.get_c_register(inputs[0])
+
+            # Need to create a singleton enum to hold the WASM function name.
+            class ExtOpName(Enum):
+                WASM = op.func_name
+
+            # Update datastructures with WASM function name and
+            # appropriate definition.
+
+            # Update translation dict.
+            _TK_TO_PYQIR[OpType.WASM] = QirGate(
+                opnat=OpNat.HYBRID, opname=ExtOpName.WASM, opspec=OpSpec.BODY
+            )
+
+            # Update gateset.
+            gateset = PYQIR_GATES.gateset
+            gateset["wasm"] = CustomQirGate(
+                opnat=OpNat.HYBRID,
+                opname=ExtOpName.WASM,
+                opspec=OpSpec.BODY,
+                function_signature=[types.Int(64)],
+                return_type=types.Int(64),
+            )
+
+            # Update gateset in module.
+            module.gateset = PYQIR_GATES
+
+            # A utility function to convert from a pytket
+            #  BitRegister to an SSA variable via pyqir types.
+            reg2var = module.module.add_external_function(
+                "reg2var", types.Function([types.BOOL] * 64, types.Int(64))
```

Comment:
Does this only work with registers of size 64? Should we also handle smaller registers by zero-padding?

Reply:
Atm, yes. I can check on the size of `reg` and zero-pad `reg_bool` if necessary.

Reply:
Done.